flink-runtime/src/main/java/org/apache/flink/runtime/management/JMXService.java

```diff
@@ -100,4 +104,28 @@ private static JMXServer startJMXServerWithPortRanges(Iterator<Integer> ports) {
 		}
 		return successfullyStartedServer;
 	}
+
+	/**
+	 * This method tries to poke for a new port number, then releases it.
+	 *
+	 * @return a successful poked port number.
+	 */
+	private static int tryPokeForNewPort() {
+		ServerSocket serverSocket = null;
+		try {
+			serverSocket = new ServerSocket(0);
+			return serverSocket.getLocalPort();
+		} catch (IOException e) {
+			LOG.warn("Unable to allocate new Server Socket!", e);
+			return -1;
```

Comment:
If we go with this approach, we need better error handling.
This logs a warning, returns -1, and the JMXServer will throw up another error later one.